ydkgen/resolver/bundle_translator.py

```diff
@@ -24,214 +24,163 @@
-            files_attr.append(attr)
-    return files_attr
-
-
-def get_remote_file_attr(url, commitid, path, tmp_dir):
-    file_attr = {}
-    file = os.path.join(tmp_dir, path)
-
-    name, revision, kind = get_attr(tmp_dir, file)
-    uri = convert_uri(url=url, commitid=commitid, path=path)
-    return {
-            'name' : name,
-            'revision' : revision,
-            'kind' : kind,
-            'uri' : uri
-            }
-
-def get_remote_files_attr(url, commitid, d, tmp_dir):
-    files_attr = []
-    target_dir = os.path.join(tmp_dir, d)
-    for (dirpath, dirnames, filenames) in walk(target_dir):
-        for filename in filenames:
-            path = os.path.join(d, dirpath, filename)
-            attr = get_remote_file_attr(url, commitid, path, tmp_dir)
-            files_attr.append(attr)
-    return files_attr
-
-def get_bundle_definition(data, file):
-    major, minor, patch = data['version'].split('.')
-    name = os.path.basename(file).split('.json')[0]
-    return {
-            "name"  : name,
-            "major" : major,
-            "minor" : minor,
-            "patch" : patch
-            }
-
-
-def get_ydk_version(data):
-    major, minor, patch = data['version'].split('.')
-    return {
-            "major" : major,
-            "minor" : minor,
-            "patch" : patch
-            }
-
-def print_json_output(file, output_dir):
+        uri = Remote_URI(remote.url, remote.commitid, rpath)
+    return Module(name, revision, kind, uri)
+
+def get_file_attrs(files, root, remote=None):
+    for f in files:
+        logger.debug('Getting attrs from file: %s' % f)
+        yield get_module_attrs(os.path.join(root, f), root, remote)
+
+def get_dir_attrs(dirs, root, remote=None):
+    for d in dirs:
+        for (d, _, files) in walk(os.path.join(root, d.lstrip('/'))):
+            for res in  get_file_attrs((os.path.join(d, f) for f in files), root, remote):
+                yield res
+
+def get_git_attrs(repos, root, remote=None):
+    for g in repos:
+        url, tmp_dir = g['url'], tempfile.mkdtemp(suffix='.yang')
+        logger.debug('Cloning from %s to %s' % (url, tmp_dir))
+        repo = Repo.clone_from(url, tmp_dir)
+        for c in g['commits']:
+            commitid = c['commitid'] if 'commitid' in c else 'HEAD'
+            repo.git.checkout(commitid)
+            if 'file' in c:
+                for fattr in get_file_attrs(c['file'], tmp_dir, Remote(url, commitid)):
+                    yield fattr
+            if 'dir' in c:
+                for fattr in get_dir_attrs(c['dir'], tmp_dir, Remote(url, commitid)):
+                    yield fattr
+        logger.debug('Removing folder %s:' % tmp_dir)
+        rmtree(tmp_dir)
+
+def check_envs():
+    if not os.environ.has_key('YDKGEN_HOME'):
+        logger.error('YDKGEN_HOME not set.')
+        print >> sys.stderr, "Need to have YDKGEN_HOME set!"
+        sys.exit(1)
 
+def populate_temple(in_file, out_file):
```

Comment:
Yes,  you're right, and the same with the line at the bottom of this file, sorry about that, will change in next commit.